elastic/elastic-agent
# Bundle elastic-agent.app for MacOS, needed to be able to enable the  …

## What does this PR do?

* Bundles elastic-agent as MacOS app bundle. This is need to be able to give Elastic Agent Full Disk Access that can be propagated down to osquery.
* Replaces the startup shell script with symlink
* Updated code in few places that relied on the os.Executable path discovery

Addresses [FDA support for Osquery Manager](https://github.com/elastic/security-team/issues/3983)

ATTN!!!: This was tested with signing the app bundle with my developer certificate. Need to figure out how to sign the app bundle with our production cert, probably requires some work with #infra team 

## Why is it important?

Addresses [FDA support for Osquery Manager](https://github.com/elastic/security-team/issues/3983)

## Checklist


## How to test this PR locally

1. Install the agent
2. Verify that it is running and healthy
2. Enable Full disk access through JAMF policy for example
3. Verify the the files in the restricted areas accessible. For example with planted test files the query 
```
SELECT f.path FROM file AS f JOIN mdfind ON mdfind.path = f.path AND mdfind.query = "kMDItemTextContent == 'testest123*'";
```
should be able to find them in the Desktop and Download locations:

<img width="1228" alt="Screen Shot 2022-07-13 at 9 54 06 AM" src="https://user-images.githubusercontent.com/872351/178757062-0eb37bfe-5d62-4c9b-be54-ca2df7263ab3.png">


## Related issues

- Closes https://github.com/elastic/security-team/issues/3983

## Screenshots

The symlink replacing the shell script wrapper:
<img width="940" alt="Screen Shot 2022-07-12 at 7 27 15 PM" src="https://user-images.githubusercontent.com/872351/178757513-a04cf90d-f48d-4154-a79e-86b10466aa9e.png">

The agent install with the app bundle (symlink points to the binary in the app bundle)
<img width="1121" alt="Screen Shot 2022-07-12 at 7 30 32 PM" src="https://user-images.githubusercontent.com/872351/178757677-ae73c586-45b2-48f1-b634-510e69f783f6.png">

Running agent:
<img width="1679" alt="Screen Shot 2022-07-12 at 7 30 49 PM" src="https://user-images.githubusercontent.com/872351/178757856-953ead4a-258e-408e-a476-c1d660d60893.png">

App bundle dir in installed agent directory:
<img width="981" alt="Screen Shot 2022-07-12 at 7 31 19 PM" src="https://user-images.githubusercontent.com/872351/178757939-7b2bd27e-8cf6-4ebd-8ebd-acb426f93eea.png">

Dev cert signed app bundle:
<img width="1685" alt="Screen Shot 2022-07-12 at 7 37 05 PM" src="https://user-images.githubusercontent.com/872351/178758027-25eb42ae-ae0f-45e4-a7c1-c613b2b0dee7.png">

JAMF profile allow-listing the agent app bundle:
<img width="661" alt="Screen Shot 2022-07-13 at 9 54 27 AM" src="https://user-images.githubusercontent.com/872351/178758138-de5a4b2f-fc0e-4051-ab23-4134d8079a8c.png">

Query results with Full Disk Access enabled:
<img width="1228" alt="Screen Shot 2022-07-13 at 9 54 06 AM" src="https://user-images.githubusercontent.com/872351/178758285-ab6c88e5-f9ed-4b31-b8aa-b4edb53c37e4.png">
